src/lib/util/feature-evaluator/unleash.ts

```diff
@@ -0,0 +1,210 @@
+import { EventEmitter } from 'events';
+import Client, { FeatureEvaluationResult } from './client';
+import Repository, { RepositoryInterface } from './repository';
+import { Context } from './context';
+import { Strategy, defaultStrategies } from './strategy';
+
+import { ClientFeaturesResponse, FeatureInterface } from './feature';
+import { Variant, getDefaultVariant } from './variant';
+import { FallbackFunction, createFallbackFunction } from './helpers';
+import {
+    BootstrapOptions,
+    resolveBootstrapProvider,
+} from './repository/bootstrap-provider';
+import { StorageProvider } from './repository/storage-provider';
+import { UnleashEvents } from './events';
+
+export { Strategy, UnleashEvents };
+
+export interface UnleashConfig {
+    appName: string;
+    environment?: string;
+    projectName?: string;
+    strategies?: Strategy[];
+    repository?: RepositoryInterface;
+    bootstrap?: BootstrapOptions;
+    bootstrapOverride?: boolean;
+    storageProvider?: StorageProvider<ClientFeaturesResponse>;
+}
+
+export interface StaticContext {
+    appName: string;
+    environment: string;
+}
+
+export class Unleash extends EventEmitter {
```

Comment:
Might be able to remove event emitter from this

Reply:
@andreas-unleash I realized we can delete some events, but not all. We still need the repository to be ready before we can use the client. However, we may be able to make that synchronous.

Reply:
Yeah, on closer inspection: I totally made it work 🥳